### ISSUE TITLE
Freshclam returning -9 error code 

### ISSUE THREAD

**Issue by rvacaru** [upsidetravel/bucket-antivirus-function#92]
Since 12th October my update lambda logs the following in CloudWatch:

```
WARNING: Your ClamAV installation is OUTDATED!
WARNING: Local version: 0.101.2 Recommended version: 0.101.4
DON'T PANIC! Read https://www.clamav.net/documents/upgrading-clamav
main.cvd is up to date (version: 58, sigs: 4566249, f-level: 60, builder: sigmgr)
Downloading daily-25600.cdiff [100%]
Downloading daily-25601.cdiff [100%]
Downloading daily-25602.cdiff [100%]
Downloading daily-25603.cdiff [100%]
Downloading daily-25604.cdiff [100%]
Downloading daily-25605.cdiff [100%]
Downloading daily-25606.cdiff [100%]
Downloading daily-25607.cdiff [100%]
Downloading daily-25608.cdiff [100%]
Downloading daily-25609.cdiff [100%]

Unexpected exit code from freshclam: -9.
Not uploading main.cvd because md5 on remote matches local.
Not uploading daily.cld because md5 on remote matches local.
Not uploading bytecode.cvd because md5 on remote matches local.
```
I've looked online for [freshclam](https://linux.die.net/man/1/freshclam) error code -9 and didn't find anything. The -9 only reminds me of SIGKILL.
The log is provided by the `update_defs_from_freshclam` function in `clamav.py`.
I am running the update lambda code from this repository from around February 2019, so I don't have the latest commits you've been merging these past months.

Would you know what this is and how to fix it?

**Comment by rvacaru** [comment]
These are our findings and how we'll proceed:
- the update lambda times out constantly, the timeout we set is 5 minutes
- the timeout is on the download of the [daily.cvd](http://mirror.webtastix.net/clamav/) file which has a size of ~50MB
- even if we delete the daily.cvd file from the s3 bucket and re run the lambda it still fails
- setting the timeout to 15 minutes also doesn't help as freshclam returns code 55 which means error while reading the virus definition db
- freshclam command has a timeout of 30 seconds when reading from the db, we believe this is why the lambda fails even with 15 minutes of time

A solution we have found to work is 
- set the memory of the update lambda to 1024MB
- now the update takes about ~1.5 minutes 

As per why before it was working fine and then it started breaking we still don't know.

**Comment by chrisgilmerproj** [comment]
Thank you for bringing this to me. Could you see if upgrading to master fixes this for you? Look at #93 and #96. Also ensure that your IAM policy is up to date. I'm going to close this but if that doesn't resolve the issue I'll be happy to reopen and investigate.

**Comment by maxpeterson** [comment]
I encountered a similar timeout issue downloading the ClamAV definitions (for the first time). Updating the lambda memory to 1024MB worked for me!
